src/dist/manifestation.rs

```diff
@@ -499,6 +472,50 @@ impl Manifestation {
 
         Ok(tx)
     }
+
+    fn install_component<'a>(
```

Comment:
This seems more strongly attached to `ComponentBinary` than to `Manifestation`, so would suggest attaching it to that type. Suggest ordering arguments from most specific to least specific, so `tx` probably comes after `installer_file` and `new_manifest` probably comes before `download_cfg`.